docs/csharp/language-reference/operators/conditional-operator.md

```diff
@@ -10,70 +10,76 @@ helpviewer_keywords:
 
-When the result is a value, the type of `first_expression` and `second_expression` must be the same, or there must be an implicit conversion from one type to the other. When the result is a `ref`, the type of `first_expression` and `second_expression` must be the same.
+## Conditional ref expression
 
-You can express calculations that might otherwise require an `if-else` construction more concisely by using the conditional operator. For example, the following code uses first an `if` statement and then a conditional operator to classify an integer as positive or negative.
+Beginning with C# 7.2, you can use the conditional ref expression to return the reference to the result of one of the two expressions. You can assign that reference to a [local ref](../keywords/ref.md#ref-locals) or [local ref readonly](../keywords/ref.md#ref-readonly-locals) variable, use as a [reference return value](../keywords/ref.md#reference-return-values) or as a [`ref` method parameter](../keywords/ref.md#passing-an-argument-by-reference).
+
+The syntax for the conditional ref expression is as follows:
 
 ```csharp
-int input = Convert.ToInt32(Console.ReadLine());  
-string classify;  
-  
-// if-else construction.  
-if (input > 0)  
-    classify = "positive";  
-else  
-    classify = "negative";  
-  
-// ?: conditional operator.  
-classify = (input > 0) ? "positive" : "negative";  
+condition ? ref consequence : ref alternative
 ```
 
-The conditional operator is right-associative. The expression `a ? b : c ? d : e` is evaluated as `a ? b : (c ? d : e)`, not as `(a ? b : c) ? d : e`.  
-  
-The conditional operator cannot be overloaded.
-  
-## Example
+Like the original conditional operator, the conditional ref expression evaluates only one of the two expressions: either `consequence` or `alternative`.
 
-The following example shows the conditional operator whose result is a value:
+In case of the conditional ref expression, the type of `consequence` and `alternative` must be the same.
 
-[!code-csharp[csRefOperators?:](~/samples/snippets/csharp/language-reference/operators/ConditionalExamples.cs#ConditionalValue)]
+The following example demonstrates the usage of the conditional ref expression:
+
+[!code-csharp[conditional ref](~/samples/snippets/csharp/language-reference/operators/ConditionalExamples.cs#ConditionalRef)]
+
+For more information, see the [feature proposal note](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/conditional-ref.md).
+
+## Conditional operator and an `if..else` statement
+
+Usage of the conditional operator over an [if-else](../keywords/if-else.md) statement might result in more concise code in cases when you need conditionally to compute a value. The following example demonstrates two ways to classify an integer as negative or nonnegative:
+
+[!code-csharp[conditional and if-else](~/samples/snippets/csharp/language-reference/operators/ConditionalExamples.cs#CompareWithIf)]
```

Comment:
The example is not building, and I have no idea why:

![image](https://user-images.githubusercontent.com/10886961/48743710-170f0d00-ec19-11e8-95b2-45c5916507cd.png)

The build is also not generating any warnings.

Reply:
@rpetrusha  I also didn't notice any warnings even before the samples PR was merged. Does it build properly after merging the samples PR?